statsd/statsd.go

```diff
@@ -489,15 +489,23 @@ func (c *Client) SimpleServiceCheck(name string, status ServiceCheckStatus) erro
 	return c.ServiceCheck(sc)
 }
 
+func (c *Client) shutdown() {
+	c.Lock()
+	defer c.Unlock()
+	select {
+	case <-c.stop:
```

Comment:
Is this needed?
It looks like it's guarding against `shutdown()` being called more than once.

Reply:
Closing a closed channel panics

Reply:
I added this initially to prevent a panic if `Close()` was called more than once.

On closer inspection, the current implementation already panics if `Close()` is called more than once, because it calls `c.sender.close()`, which unconditionally closes `c.sender`'s stop channel.

I'll refactor this.

Reply:
> On closer inspection, the current implementation already panics if `Close()`
In my opinion this is not a reason to continue that trend and I prefer your version. An instrumentation library should never crash a host application, regardless of user error. I'd make sure `Close` never panics.